docs/upgrade.html

```diff
@@ -66,12 +66,23 @@
         individual messages is only reduced by the overhead of the batch format. This similarly affects the
         producer's <code>batch.size</code> configuration.</li>
     <li>GC log rotation is enabled by default, see KAFKA-3754 for details.</li>
+    <li>Message protocol introduces support for headers as an array of Header with uftf8 <code>key</code> and byte[] <code>value</code>. 
+        Header's will be available once the broker is upgraded to the latest protocol. Older clients will not be able to access the headers as they don't exist in previous protocols as such the broker removes the header on protocol downgrade.</li>
+    <li>ProducerRecord and ConsumerRecord client API's expose the new Headers API via <code>Headers headers()</code> method call. 
```

Comment:
So it seems i just need to thin out/be less verbose (prob just take @becketqin improved version). And just add versions on the protocol. will do this.

Javadocs all there already, so should be picked up.